presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -149,27 +153,31 @@
     private final HiveCluster clientProvider;
     private final Function<Exception, Exception> exceptionMapper;
     private final boolean impersonationEnabled;
+    private final HiveMetastoreAuthentication metastoreAuthentication;
```

Comment:
Looks like this instance is not used and we only care if it is a NoHiveMetastoreAuthentication. In this case, could we make it a boolean?

Reply:
> Looks like this instance is not used and we only care if it is a NoHiveMetastoreAuthentication. In this case, could we make it a boolean? 
> 

I also want do this. But metastore auth parameter(hive.metastore.authentication.type) is in HiveClientConfig instead of MetastoreClientConfig, in this case i can not get this parameter in presto-hive-metastore module. 
 So I think we need to introduce this NoHiveMetastoreAuthentication instance to check whether authentication is enabled anyway.  wdyt? thx